base/segment.py

```diff
@@ -165,7 +167,7 @@ def by(**type):
     listable = [item for item in __iterate__(**type)]
     if len(listable) > 1:
         maxaddr = max(builtins.map(interface.range.end, listable) if listable else [1])
-        caddr = math.floor(1. + math.log(maxaddr or 1) / math.log(16))
+        caddr = math.ceil(math.log(maxaddr or 1, 16))   # only works with powers of 16
```

Comment:
not too important, but this should mention that it only works with powers of 2 and related.